### ISSUE TITLE
`multiline-expression-wrapping` collision with `chain-wrapping`

### ISSUE THREAD

**Issue by SimonMarquis** [pinterest/ktlint#2283]
## Expected Behavior

As per the documentation on [chain wrapping](https://pinterest.github.io/ktlint/1.0.0/rules/standard/#chain-wrapping), this style is valid:

```kotlin
val foo = listOf(1, 2, 3)
    .filter { it > 2 }!!
    .takeIf { it.count() > 100 }
    ?.sum()
```

## Observed Behavior

Running `./gradlew spotlessCheck` (with `ktlint_official` version 1.0.0) wants to reformat it like that:

```
val foo =
    listOf(1, 2, 3)
        .filter { it > 2 }!!
        .takeIf { it.count() > 100 }
        ?.sum()
```

Manually disabling `standard:multiline-expression-wrapping` (and dependant `standard:string-template-indent`) makes the initial formatting valid again.

From a user perspective, this feels like an unexpected behaviour, but is it on purpose?

I only found this potentially relevant issue:
- https://github.com/pinterest/ktlint/issues/2183

Should this be documented in the chain wrapping docs or is it a bug?

## Your Environment

* Version of ktlint used: 1.0.0
* Version of spotless used: 6.22.0
* Relevant parts of the `.editorconfig` settings
  ```properties
  [*.{kt,kts}]
  ktlint_code_style = ktlint_official
  ```
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): `./gradlew spotlessCheck`
* Version of Gradle used (if applicable): 8.3
* Operating System and version: macOS 13.5.2

**Comment by paul-dingemans** [comment]
> Manually disabling `standard:multiline-expression-wrapping` (and dependant `standard:string-template-indent`) makes the initial formatting valid again.
> 
> From a user perspective, this feels like an unexpected behaviour, but is it on purpose?
> ...
> Should this be documented in the chain wrapping docs or is it a bug?

As always, it depends. It is merely impossible to document all nuances of each rule. Especially when rules can be impacted by other rules. Also, I have not yet found a way to ensure that all examples in the documentation are still valid after adding a new rule.

So, in this case the example of `chain-wrapping` rule is valid when rule `multiline-expression-wrapping` is disabled. The latter rule is enabled by default in code style `ktlint_official` only. If you have not set `.editorconfig` property `ktlint_code_style` than the migration from ktlint `0.x` to `1.x` effectively changed your code style from `intellij_idea` to `ktlint_official`. See https://pinterest.github.io/ktlint/1.0.0/rules/code-styles/

As `ktlint_official` is now the default code style, the `multiline-expression-wrapping` is enabled by default as well. As of that the code sample of the `chain-wrapping` rule is misleading.

**Comment by paul-dingemans** [comment]
I have found quite some more examples in the documentation where the examples with allowed code did not comply with ktlint code style `ktlint_official` (including experimental rules). They will be fixed in PR #2285 and published with `1.0.1` release.
